### PR TITLE
Cleanup to reduce diff

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -57,10 +57,11 @@ if( ${TARGET_OS} STREQUAL "Linux" )
 elseif(${TARGET_OS} STREQUAL "Windows" )
   set(OS "WINDOWS")
   set(OSNAME "Windows")
+  add_definitions(-DWIN32 -DHOST_WIN -DWINNT -DTARGET_WIN -DTARGET_WIN_X86)
   if( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-    add_definitions(-DWIN32 -DHOST_WIN -DWIN64 -DWINNT -DTARGET_WIN_X8632)
+    add_definitions(-DWIN64 -DTARGET_WIN_X8664)
   else( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-    add_definitions(-DWIN32 -DHOST_WIN -DWIN64 -DWINNT -DTARGET_WIN_X8632)
+    add_definitions(-DTARGET_WIN_X8632)
```
Comment (isuruf) [inline CMakeLists.txt:64]: @xoviat, can you have a look above?

Comment (ghost) [reply CMakeLists.txt:64]: It seems to be okay.

**Comment by ghost** [comment]
```
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     323  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 224)
  0 inform,   0 warnings,   1 severes, 0 fatal for ieee_get_flag
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     323  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 231)
  0 inform,   0 warnings,   1 severes, 0 fatal for ieee_get_flag_l8
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     326  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 254)
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     323  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 256)
  0 inform,   0 warnings,   2 severes, 0 fatal for ieee_set_flag_scalar
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     324  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 272)
  0 inform,   0 warnings,   1 severes, 0 fatal for ieee_set_flag_array
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     324  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 287)
  0 inform,   0 warnings,   1 severes, 0 fatal for ieee_set_flag_arrscal
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     323  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 294)
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     326  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 296)
  0 inform,   0 warnings,   2 severes, 0 fatal for ieee_set_halting_mode_scalar
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     324  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 313)
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     327  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 314)
  0 inform,   0 warnings,   2 severes, 0 fatal for ieee_set_halting_mode_array
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     324  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 330)
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     327  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 331)
  0 inform,   0 warnings,   2 severes, 0 fatal for ieee_set_halting_mode_arrscal
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     326  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 339)
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     323  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 341)
  0 inform,   0 warnings,   2 severes, 0 fatal for ieee_set_flag_scalar_l8
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     324  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 357)
  0 inform,   0 warnings,   1 severes, 0 fatal for ieee_set_flag_array_l8
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     324  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 372)
  0 inform,   0 warnings,   1 severes, 0 fatal for ieee_set_flag_arrscal_l8
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     323  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 379)
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     326  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 381)
  0 inform,   0 warnings,   2 severes, 0 fatal for ieee_set_halting_mode_scalar_l8
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     324  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 398)
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     327  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 399)
  0 inform,   0 warnings,   2 severes, 0 fatal for ieee_set_halting_mode_array_l8
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     324  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 415)
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     327  (C:\projects\flang\runtime\flang\ieee_exceptions.F95: 416)
  0 inform,   0 warnings,   2 severes, 0 fatal for ieee_set_halting_mode_arrscal_l8
F90-S-0000-Internal compiler error. insert_argrsrv: no reg args     326  (C:\projects\flang\r
```

This is currently beyond me at this point.

File: CMakeLists.txt (hunk 1 above)
Comment (ghost) [reply CMakeLists.txt:64]: Actually I think `TARGET_WIN` is the problem.

**Comment by isuruf** [comment]
It's because of `TARGET_WIN_X8664` in `tools/flang2/flang2exe/iliutil.c`

**Comment by ghost** [comment]
```
__fmth_i_dmod referenced in function __mth_i_dmod
error.c.obj : error LNK2019: unresolved external symbol __fortio_setmode_binary referenced in function __fortio_init
```

**Comment by ghost** [comment]
I thought we already had __fortio_setmode_binary?

**Comment by isuruf** [comment]
No, that was `__fortio_binary_mode`

**Comment by isuruf** [comment]
@xoviat, are you sure about whether `__fortio_binary_mode` does the right thing?

**Comment by ghost** [comment]
No. Maybe it's asking for whether the steam is binary? I assumed it was setting it.

**Comment by isuruf** [comment]
Yes, that's what I thought too since `__fortio_setmode_binary` is there.

**Pull request merged**